### PR TITLE
OCPBUGS-11801: Fix agent-tui libnmstate dependency name

### PR DESCRIPTION
**Pull request by andfasano**
This patch fixes the https://issues.redhat.com/browse/OCPBUGS-11801 bug by:
* removing the hard-coded version when extracting libnmstate
*  copying also the symlinks in the agent cpio archive

Due a recent bump in the repos, the agent-tui (currently shipped in the `agent-installer-node-agent` image) is now comprised of the following files:
```
agent-tui
libnmstate.so.2 -> libnmstate.so.2.2.9 (symlink)
libnmstate.so.2.2.9
```

### CODE REVIEW

File: pkg/asset/agent/image/agentimage.go
Comment (bfournie) [inline pkg/asset/agent/image/agentimage.go:80]: I think it can be done in a more generic way and still keep the wildcard at line 77. We should still support libnmstate.1 until the CI builds are updated and also libnmstate.so.2.2.10 is available soon.

For example, perhaps use
libnmstateFull := "libnmstate.so.2.*"
libnmstateOveride := "libnmstate.so.2"

And keep agentTuiFilenames the same at line 77 with the wildcard.
Then at line 94 do a Glob match on libnmstateFull and only override in that case.

Comment (andfasano) [reply pkg/asset/agent/image/agentimage.go:80]: I'm not convinced that the effort to support libnmstate.1 is worth the game, considering that it's present only in the CI payload for mistake (given that it's  hasn't been yet updated).
The glob on the minor version seems a good idea I'll update it.

**Comment by bfournie** [comment]
I've tested the fix in dev-scripts and it looks good.  I was able to install the latest nightly, see the TUI during boot, and ssh in in bootstrap.

$ ls -al /usr/local/bin/libnm*
-r-xr-xr-x. 1 root root 7990432 Apr 14 14:25 /usr/local/bin/libnmstate.so.2

Note that the agent CI tests will continue to fail until `agent-installer-utils` is updated to use libnmstate.2 in CI.

/approve

**Comment by andfasano** [comment]
/hold 

We decided to update assisted-installer-agent before landing this patch, so that libnmstate.2 will be available also in the CI stream

**Comment by andfasano** [comment]
Requires https://github.com/openshift/assisted-installer-agent/pull/531

File: pkg/asset/agent/image/agentimage.go
Comment (patrickdillon) [inline pkg/asset/agent/image/agentimage.go:104]: nit: IMHO it is a little easier to read:

```go
		override := filepath.Join(filepath.Dir(f), libnmstateOverride)
		err = os.Rename(f, override)
		if err != nil {
			return nil, err
		}
```
or even

```go
		override := filepath.Join(filepath.Dir(f), libnmstateOverride)
		if err = os.Rename(f, override); err != nil {
			return nil, err
		}
```

File: pkg/asset/agent/image/agentimage.go
Comment (patrickdillon) [inline pkg/asset/agent/image/agentimage.go:96]: In general we should consider [wrapping these errors](https://pkg.go.dev/github.com/pkg/errors#Wrapf) to make it easier to discern failure points within the same function

**Review by patrickdillon — COMMENTED** [review]
The code I here LGTM, but I'm still not as familiar with the agent code as I would like it and it is unclear to me why renaming the file to `libnmstate.so.2` resolves this bug.

**Comment by rwsu** [comment]
> The code I here LGTM, but I'm still not as familiar with the agent code as I would like it and it is unclear to me why renaming the file to `libnmstate.so.2` resolves this bug.

I think it is because agent-tui is expecting to load libnmstate.so.2, while in the release image the file is actually named libnmstate.so.2.2.9. Normally there would be a symlink from libnmstate.so.2 to libnmstate.so.2.2.9 when the nmstate-libs package is installed, but here we are copying the library file from the release image and inserting it into the agent ISO with the "correct" name. Perhaps a note in `fetchAgentTuiFiles` explaining why the file is being renamed would help.

**Comment by rwsu** [comment]
lgtm. I suggest adding a comment to explain the file renaming.

**Comment by andfasano** [comment]
After a discussion with @zaneb, I've switched the implementation so that now also the symbolic links extracted are copied into the agent ISO (while before they were ignored)

File: pkg/asset/agent/image/cpio.go
Comment (zaneb) [inline pkg/asset/agent/image/cpio.go:98]: nit: could move the `Open()` call to here.

**Review by zaneb — COMMENTED** [review]
/lgtm

File: pkg/asset/agent/image/cpio.go
Comment (andfasano) [reply pkg/asset/agent/image/cpio.go:98]: Thanks, good point

**Comment by pawanpinjarkar** [comment]
/lgtm
Please update the description mentioning the details about the fix.

**Review by rwsu — APPROVED** [review]

**Pull request merged**
